src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -591,6 +591,13 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                     }
                 }
 
+                if (results.Count > 0 && ((CSharpCompilation)_compilation).LanguageVersion < LanguageVersion.CSharp8)
```

Comment:
Nit: I think it's only possible for compilations to be `CSharpCompilation` instances in source gen, but perhaps you could try avoiding the hard casts by either having the parser doing it for you at construction or use an `_compilation is CSharpCompilation { LanguageVersion : < LanguageVersion.CSharp8 }` check.